### ISSUE TITLE
collapse_vars generates invalid code

### ISSUE THREAD

**Issue by neelance** [swc-project/swc#6217]
### Describe the bug

The collapse_vars transformation generates invalid code, because it moves the initialisation of a variable below its use.

Breaks https://github.com/mozilla/pdf.js/

### Input code

```typescript
const a = 0;
let b;
for (let i = 0; i < 2; ++i) {
  if (i === 0) {
    b = 1;
  } else {
    b = 2;
  }
  console.log(a, b);
}

let c = 1, d;
while (c--) { }
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript",
      "jsx": false
    },
    "target": "es2019",
    "loose": false,
    "minify": {
      "compress": {
        "arguments": false,
        "arrows": false,
        "booleans": false,
        "booleans_as_integers": false,
        "collapse_vars": true,
        "comparisons": false,
        "computed_props": false,
        "conditionals": false,
        "dead_code": false,
        "directives": false,
        "drop_console": false,
        "drop_debugger": false,
        "evaluate": false,
        "expression": false,
        "hoist_funs": false,
        "hoist_props": false,
        "hoist_vars": false,
        "if_return": false,
        "join_vars": false,
        "keep_classnames": false,
        "keep_fargs": false,
        "keep_fnames": false,
        "keep_infinity": false,
        "loops": false,
        "negate_iife": false,
        "properties": false,
        "reduce_funcs": false,
        "reduce_vars": false,
        "side_effects": false,
        "switches": false,
        "typeofs": false,
        "unsafe": false,
        "unsafe_arrows": false,
        "unsafe_comps": false,
        "unsafe_Function": false,
        "unsafe_math": false,
        "unsafe_symbols": false,
        "unsafe_methods": false,
        "unsafe_proto": false,
        "unsafe_regexp": false,
        "unsafe_undefined": false,
        "unused": false,
        "const_to_let": false,
        "pristine_globals": false
      },
      "mangle": false
    }
  },
  "module": {
    "type": "es6"
  },
  "minify": false,
  "isModule": true
}
```

### Playground link

https://play.swc.rs/?version=1.3.9&code=H4sIAAAAAAAAA02NvQ6DMAyEdz%2FFjUH8CFhdHoakBiJZjVSQOlR5d5x06WD59J19F9LrvLBiwcikcsEzbekNV3Ss2NYDM6NtY4MvAXGDM2sx8wcAb5cTm8wQPeWPzpXaBGtKKoOm3a0dfMOUqVaG8tzhyfQ5ogpc6HsLRr4BOEe3%2B50AAAA%3D&config=H4sIAAAAAAAAA22TwXLjIAyG38XnHHZ76Mz2AfbWZ2AICIcWkAeJNJ5M3r0icQ2xe0Mfv%2BVfQroOH2SGt%2Bsw6UyQ64nmxPoyvA1goiaT%2FcTDQWSCnA4Et8PAOo%2FAVUIvf%2F7%2Bk%2BuASLAIDkP0ybu5JjMYpwxE9SwflQiJadXpnPGrhUfEADrtgdKkfGIYIbdLgyHoiUCddaWcyx1GqcQTpl4Yp8Jg1ZRx6nGynj0mCVdoQVtl0LZarM9g2J%2BhE0keESUSd8%2FQwrGMY%2B3jQuGsQ9HcZHC590N%2Bu6ITemLlSmf5gZ79Ptij2AV5pzJwyS3XB%2Fr0rPkEELNBEyUdYcOdvMkW%2FSLzycmL8rxSee7OWYJRSlTeu1Zn9Q6ZfZcqgy0GaqFmB58sk7egwDnpewe%2FPJtTl47nCdC1WPqnOwOPUG1GbKF1JHbwvxjj%2FmEWHjWftozmeMSwSxGBT2h3WJrBuIVZxvkybWlJFqTZYLuLQl1Y544Vowp1A9duy2zIV2oMeGzzLKsadRrXKb1VgLZUcL038L7Dr8Ot7ezPaNH7IqyLdfsGbc%2FDZCgEAAA%3D

### Expected behavior

Generates valid code.

### Actual behavior

Generates code that causes "Uncaught ReferenceError: Cannot access 'b' before initialization" error.

### Version

1.3.9

**Comment by kdy1** [comment]
Seems like it's already fixed on unpublsihed code https://github.com/swc-project/swc/pull/6218

**Comment by neelance** [comment]
@kdy1 This is supposed to be fixed with v1.3.10, right? But when I open the playground link above and select 1.3.10, then it is still broken. What am I missing?

**Comment by remorses** [comment]
I think my Next.js app is having a lot of `ReferenceError` because of this issue, when will the fix be available in Next?

**Comment by BPScott** [comment]
@neelance it looks like the fix went out in 1.3.11

**Comment by neelance** [comment]
I can confirm that it has fixed the issue for us. Thanks! 🙏
